### PR TITLE
Align the encryption flags

### PR DESCRIPTION
**Pull request by ReToCode**
# Changes
- Align the encryption flags according to https://github.com/knative/serving/issues/14368
- Adds reverse compatibility for existing flags
- Drop `dataplane-trust` and `controlplane-trust`

I'd argue that its ok to drop the dataplane-& controlplane-trust flags, as they are clearly marked as ALPHA. IMHO, Introducing backward compatibility for those would make it really unread and unmaintainable. Also we have not implemented this feature and various values of those, so it's basically not possible to be reverse compatible.

/kind cleanup

Partially fixes https://github.com/knative/serving/issues/14368

**Release Note**

```release-note
Renames the flags that control encryption
- `auto-tls` is now named `external-domain-tls`
- `internal-encryption` is now named `system-internal-tls`
- `cluster-local-domain-tls` is introduced as a new alpha state flag to control TLS certificates for cluster-local domains
```

**Docs**
PRs for docs will follow separately in https://github.com/knative/serving/issues/14368

### CODE REVIEW

**Comment by ReToCode** [comment]
About the expected CI errors:

I see the following options
- Keep the dropped flags here (without handling the compatibility stuff), rework all repos, then drop them here in a new PR
- YOLO - ignore it for now, fix it after follup-PRs on Serving + net-contour are through

We'd need to do both approaches our configuration would be broken for a short time period. And we'd need to do the fixes in a short time period and before the release.

Opinions?

**Comment by KauzClay** [comment]
I think I could go for 
>  ignore it for now, fix it after follup-PRs on Serving + net-contour are through

I can get some quick PRs for serving and net-contour to use your new stuff ready

**Comment by nak3** [comment]
> I'd argue that its ok to drop the dataplane-& controlplane-trust flags, as they are clearly marked as ALPHA. IMHO, Introducing backward compatibility for those would make it really unread and unmaintainable. Also we have not implemented this feature and various values of those, so it's basically not possible to be reverse compatible.

I would like to get approval from TOC just in case. Otherwise LGTM.

**Comment by ReToCode** [comment]
@knative/technical-oversight-committee @dprotaso  Could you please take a look: https://github.com/knative/networking/pull/858#issuecomment-1724720424

**Comment by dprotaso** [comment]
> I would like to get approval from TOC just in case.

TOC wouldn't make this call - just leads of the WG. It's an alpha feature so I don't we should worry about the renaming.

File: pkg/config/config.go [hunk 1]
```diff
@@ -134,39 +143,26 @@ const (
...
-	// DataplaneTrustKey is the name of the configuration entry
-	// defining the level of trust used for data plane traffic.
-	DataplaneTrustKey = "dataplane-trust"
-
-	// ControlplaneTrustKey is the name of the configuration entry
-	// defining the level of trust used for control plane traffic.
-	ControlplaneTrustKey = "controlplane-trust"
+	// KnativeInternalTLSKey is the name of the configuration whether
+	// knative internal traffic is encrypted or not.
+	KnativeInternalTLSKey = "knative-internal-tls"
```
[the remark this answers is not in the document]
Comment (ReToCode) [reply pkg/config/config.go:153]: Hm I get the `knative` part, but I'm not super convinced with the `component` part (it seems a bit generic, which components?). Maybe something like this?

- `internal-communication-tls`
- `system-internal-tls`

[the remark this answers is not in the document]
Comment (KauzClay) [reply pkg/config/config.go:153]: I think I like `system-internal-tls`

[the remark this answers is not in the document]
Comment (dprotaso) [reply pkg/config/config.go:153]: Cool let's use `system-internal-tls`

[the remark this answers is not in the document]
Comment (ReToCode) [reply pkg/config/config.go]: Great, updated it to `system-internal-tls`.
@KauzClay mind updating your two WIP PRs before we merge this here?

**Comment by dprotaso** [comment]
/lgtm
/approve
/hold

@ReToCode will let you unhold

**Comment by ReToCode** [comment]
/unhold

@dprotaso can you override the failed checks?

**Pull request merged**